api/v1alpha1/gateway_types.go

```diff
@@ -139,7 +169,7 @@ const (
 	TLS1_3 = "TLS1_3"
 )
 
-// ListenerTLS describes the TLS configuration for a given port.
+// TLSConfig describes configuration for establishing a TLS connection.
```

Comment:
To me, "establishing a TLS connection" implies pretty strongly that it is being used by a network client.

Reply:
Maybe `TLSConfig` => `TLSServerConfig` will be less ambiguous/

(the server also is part of connection establishment)

Reply:
If this is intended to be specific to accepting a TLS session, then I agree `TLSServerConfig` is less ambiguous.

Reply:
@bowei I considered `TLSServerConfig` but `TLSConfig` contains client-side properties such as `CACertificates` used by the `Gateway` for establishing a TLS connection to a backend `Service`.